src/live.asp.net/gulpfile.js

```diff
@@ -1,63 +1,103 @@
+    path.dirname(require.resolve('bootstrap/dist/fonts/glyphicons-halflings-regular.woff')) + '/**',
+    // declare each file
+    require.resolve('jquery/dist/jquery.js'),
+    require.resolve('jquery/dist/jquery.min.js'),
+    // only one file is distributed
+    require.resolve('jquery-validation/dist/jquery.validate.js')
+  ]
+}
+
+paths.library = paths.webroot + library.destination;
 paths.js = paths.webroot + "js/**/*.js";
 paths.minJs = paths.webroot + "js/**/*.min.js";
 paths.css = paths.webroot + "css/**/*.css";
 paths.minCss = paths.webroot + "css/**/*.min.css";
 paths.concatJsDest = paths.webroot + "js/site.min.js";
 paths.concatCssDest = paths.webroot + "css/site.min.css";
 
-gulp.task("clean:js", function (cb) {
-    rimraf(paths.concatJsDest, cb);
+
+paths.library = paths.webroot + library.destination;
+paths.js = paths.webroot + "js/**/*.js";
+paths.minJs = paths.webroot + "js/**/*.min.js";
+paths.css = paths.webroot + "css/**/*.css";
+paths.minCss = paths.webroot + "css/**/*.min.css";
+paths.concatJsDest = paths.webroot + "js/site.min.js";
+paths.concatCssDest = paths.webroot + "css/site.min.css";
+
+gulp.task("lib", ["clean"], function () {
+  return gulp.src(library.source, { base: library.base })
+    .pipe(gulp.dest(paths.library));
+});
+
+gulp.task("clean:lib", function () {
+    return del(paths.library);
+});
+
+gulp.task("clean:js", function () {
+    return(del(paths.concatJsDest));
 });
 
-gulp.task("clean:css", function (cb) {
-    rimraf(paths.concatCssDest, cb);
+gulp.task("clean:css", function () {
+    return(del(paths.concatCssDest));
 });
 
-gulp.task("clean", ["clean:js", "clean:css"]);
+gulp.task("clean", ["clean:js", "clean:css", "clean:lib"]);
 
-gulp.task("min:js", ["clean:js", "jshint"], function () {
+gulp.task("min:js", function () {
     return gulp.src([paths.js, "!" + paths.minJs], { base: "." })
         .pipe(concat(paths.concatJsDest))
         .pipe(uglify())
         .pipe(gulp.dest("."));
 });
 
-gulp.task("min:css", ["clean:css", "csslint"], function () {
+gulp.task("min:css", function () {
     return gulp.src([paths.css, "!" + paths.minCss])
         .pipe(concat(paths.concatCssDest))
         .pipe(cssmin())
         .pipe(gulp.dest("."));
 });
 
-gulp.task("min", ["min:js", "min:css"]);
+gulp.task("min", ["lib", "csslint", "jshint", "min:js", "min:css"]);
 
-gulp.task("jshint", function() {
-    return gulp.src(paths.js)
+gulp.task("jshint", ["lib"], function() {
+    return gulp.src([paths.js, "!" + paths.minJs])
         .pipe(jshint())
-        .pipe(jshint.reporter());
+        .pipe(jshint.reporter())
 });
 
-gulp.task("csslint", function() {
-    return gulp.src(paths.css)
+gulp.task("csslint", ["lib"], function() {
+    return gulp.src([paths.css, "!"  + paths.minCss])
         .pipe(csslint())
         .pipe(csslint.reporter());
 });
 
-gulp.task("lint", ["jshint", "csslint"]);
+gulp.task("prepublish", ["lib", "csslint", "jshint",  "min"]);
```

Comment:
a couple reasons. First, changed to ensure to return streams where appropriate and second added dependencies to each. By doing so this makes sure that the tasks run sequentially and not in parallel.

docs -> https://github.com/gulpjs/gulp/blob/master/docs/recipes/running-tasks-in-series.md